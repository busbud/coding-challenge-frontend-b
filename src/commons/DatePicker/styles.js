import styled from "styled-components";

import { DatePicker } from "antd";

export const StyledDatePicker = styled(DatePicker)`
  height: 44px !important;
  border-radius: 4px !important;
  padding: 7px 12px !important;
`;
