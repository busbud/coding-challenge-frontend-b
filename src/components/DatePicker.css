.DayPickerInput{
    display:block!important;

}
.DayPicker {
    font-size: 16px;
}
