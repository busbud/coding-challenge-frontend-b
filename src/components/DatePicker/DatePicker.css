.datePicker {
  width: 100%;
  display: flex;
}
