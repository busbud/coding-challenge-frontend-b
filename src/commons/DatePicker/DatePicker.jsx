import React from "react";

import { StyledDatePicker } from "./styles";

const DatePicker = ({ ...otherProps }) => <StyledDatePicker {...otherProps} />;

export default DatePicker;
