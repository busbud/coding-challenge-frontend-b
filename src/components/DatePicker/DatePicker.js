// @flow
import React from 'react';
import styles from './DatePicker.css';

export const DatePicker = () => (
  <div className={styles.DatePicker}>
    <p>HEllo</p>
  </div>
);
