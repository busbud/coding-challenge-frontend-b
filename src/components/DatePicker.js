import React from 'react';
import { useTranslation } from 'react-i18next';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import 'react-day-picker/lib/style.css';
import './DatePicker.css'
import { useDataProvider } from '../contexts/SearchContext';

const WEEKDAYS_SHORT = {
    ru: ['Вс', 'Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб'],
    rf: 'di_lu_ma_me_je_ve_sa'.split('_'),
    en: 'Su_Mo_Tu_We_Th_Fr_Sa'.split('_'),
};
const MONTHS = {
    ru: [
        'Январь',
        'Февраль',
        'Март',
        'Апрель',
        'Май',
        'Июнь',
        'Июль',
        'Август',
        'Сентябрь',
        'Октябрь',
        'Ноябрь',
        'Декабрь',
    ],
    en: 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_'),
    fr: 'janvier_février_mars_avril_mai_juin_juillet_août_septembre_octobre_novembre_décembre'.split('_')
};
const FIRST_DAY_OF_WEEK = {
    ru: 1,
};
export const DatePicker = () => {
    const dataProv = useDataProvider();
    const { t, i18n } = useTranslation();
    return <div><DayPickerInput name="date"
        locale={i18n.language}
        months={MONTHS[i18n.language]}
        weekdaysShort={WEEKDAYS_SHORT[i18n.language]}
        firstDayOfWeek={FIRST_DAY_OF_WEEK[i18n.language]}
        onDayChange={
            day => dataProv.dispatch({
                type: "SET_SEARCH_PARAM",
                name: "date",
                value: day.toISOString().slice(0, 10)
            })
        } className="Search__input" placeholder={t("Departure date")} format="YYYY-MM-DD" />
    </div>

}
