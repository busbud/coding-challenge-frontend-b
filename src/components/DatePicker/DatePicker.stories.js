// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { DatePicker } from './DatePicker';

storiesOf('DatePicker', module)
  .add('Default', () => <DatePicker isErrored={false} label="label" onChange={action('changed')} />)
  .add('withErrors', () => <DatePicker isErrored label="label" onChange={action('changed')} />);
