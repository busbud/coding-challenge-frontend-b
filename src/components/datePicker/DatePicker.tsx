import React from 'react';
import { DatePicker } from 'antd';
import moment from 'moment';

import { DATE_FORMAT_LIST } from './DatePicker.constants';
import type { DatePickerProps, DateType } from './DatePicker.types';
import { disabledDate } from './DatePicker.utils';
import styles from './DatePicker.styles.scss';

const CustomDatePicker: React.FC<DatePickerProps> = ({
    date,
    onDateChange,
}): React.ReactElement => {
    const onChange = (value: DateType) => {
        onDateChange(value?.format(DATE_FORMAT_LIST));
    };

    return (
        <DatePicker
            className={styles.datePicker}
            size="large"
            format={DATE_FORMAT_LIST}
            disabledDate={disabledDate}
            onChange={onChange}
            value={moment(date)}
        />
    );
};

export default CustomDatePicker;
