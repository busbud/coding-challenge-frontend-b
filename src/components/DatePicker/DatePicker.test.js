import React from 'react';

import renderer from 'react-test-renderer';
import { DatePicker } from './DatePicker';

describe('DatePicker', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DatePicker onChange={() => {}} label="label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
