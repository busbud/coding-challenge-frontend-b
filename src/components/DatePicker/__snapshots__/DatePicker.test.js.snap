// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders correctly 1`] = `
<div
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
>
  <label
    className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12 MuiInputLabel-outlined-15"
    data-shrink={true}
  >
    label
  </label>
  <div
    className="MuiInputBase-root-36 MuiOutlinedInput-root-23 MuiInputBase-fullWidth-45 MuiInputBase-formControl-37 MuiInputBase-adornedStart-40 MuiOutlinedInput-adornedStart-26"
    onClick={[Function]}
    shrink={1}
  >
    <fieldset
      aria-hidden={true}
      className="MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="MuiPrivateNotchedOutline-legend-54"
        style={
          Object {
            "width": 0,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
    <div
      className="MuiInputAdornment-root-55 MuiInputAdornment-positionStart-57"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-59"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <input
      aria-invalid={false}
      className="MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputType-49 MuiInputBase-inputAdornedStart-51 MuiOutlinedInput-inputAdornedStart-34"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="date"
    />
  </div>
</div>
`;
